examples/platform/silabs/efr32/rs911x/rsi_if.c

```diff
@@ -586,27 +573,16 @@ static void wfx_rsi_do_join(void)
         /* Try to connect Wifi with given Credentials
          * untill there is a success or maximum number of tries allowed
          */
-        while (is_wifi_disconnection_event || wfx_rsi.join_retries <= WFX_RSI_CONFIG_MAX_JOIN)
+        if ((status = rsi_wlan_connect_async((int8_t *) &wfx_rsi.sec.ssid[0], connect_security_mode, &wfx_rsi.sec.passkey[0],
+                                             wfx_rsi_join_cb)) != RSI_SUCCESS)
         {
-            /* Call rsi connect call with given ssid and password
-             * And check there is a success
-             */
-            if ((status = rsi_wlan_connect_async((int8_t *) &wfx_rsi.sec.ssid[0], connect_security_mode, &wfx_rsi.sec.passkey[0],
-                                                 wfx_rsi_join_cb)) != RSI_SUCCESS)
-            {
-
-                wfx_rsi.dev_state &= ~WFX_RSI_ST_STA_CONNECTING;
-                SILABS_LOG("%s: rsi_wlan_connect_async failed with status: %02x on try %d", __func__, status, wfx_rsi.join_retries);
-
-                wfx_retry_interval_handler(is_wifi_disconnection_event, wfx_rsi.join_retries);
-                wfx_rsi.join_retries++;
-            }
-            else
-            {
-                SILABS_LOG("%s: starting JOIN to %s after %d tries\n", __func__, (char *) &wfx_rsi.sec.ssid[0],
-                           wfx_rsi.join_retries);
-                break; // exit while loop
-            }
+            wfx_rsi.dev_state &= ~WFX_RSI_ST_STA_CONNECTING;
+            SILABS_LOG("%s: rsi_wlan_connect_async failed with status: %02x on try %d", __func__, status, wfx_rsi.join_retries);
+            wfx_retry_connection(++wfx_rsi.join_retries);
+        }
+        else
+        {
+            SILABS_LOG("%s: starting JOIN to %s after %d tries\n", __func__, (char *) &wfx_rsi.sec.ssid[0], wfx_rsi.join_retries);
```

Comment:
Avoid using SILABS_LOGS.

Reply:
@mkardous-silabs all the logging will be fixed as a separate PR which is currently being worked on